### PR TITLE
add space check in StorageFindSector

### PR DESCRIPTION
**Pull request by yhio**
add space check in StorageFindSector when allowfetch

### CODE REVIEW

File: stores/index.go [hunk 1]
```diff
@@ -82,15 +82,18 @@ type storageEntry struct {
...
+
 	*indexLocks
 	lk sync.RWMutex
 
 	sectors map[Decl][]*declMeta
 	stores  map[ID]*storageEntry
 }
 
-func NewIndex() *Index {
+func NewIndex(spt abi.RegisteredSealProof) *Index {
```
Comment (magik6k) [inline stores/index.go:94]: It's better to add this param to `StorageFindSector`, we eventually want to support one sector index / storage for multiple miner actors, with multiple sector sizes

**Review by magik6k — COMMENTED** [review]
Thanks for the PR. Just 1 comment, looks good otherwise!

File: stores/index.go (hunk 1 above)
Comment (yaohcn) [reply stores/index.go:94]: i want to add param at first, but many places call StorageFindSector, it not easy pass spt param to caller

**Comment by magik6k** [comment]
Sorry for doing this in the middle of your PR, but this should make everyones lives easier.

We've moved code in this repo into lotus, to `extern/sector-storage` on the next branch. Could you reopen this PR there?

**Comment by yhio** [comment]
closing in favor of https://github.com/filecoin-project/lotus/pull/2976

**Pull request closed without merging**